input/postgres/explain_analyze.go

```diff
@@ -53,16 +58,54 @@ func runExplainAnalyze(ctx context.Context, db *sql.DB, query string, parameters
 }
 
 func validateQuery(query string) error {
-	var isUtil []bool
-	// To be on the safe side never EXPLAIN a statement that can't be parsed,
-	// or multiple statements in one (leading to accidental execution)
-	isUtil, err := util.IsUtilityStmt(query)
-	if err != nil || len(isUtil) != 1 || isUtil[0] {
-		err = fmt.Errorf("query is not permitted to run (multi-statement or utility command?)")
+	parseResult, err := pg_query.Parse(query)
+	if err != nil {
+		return fmt.Errorf("query is not permitted to run - failed to parse")
+	}
+	if len(parseResult.Stmts) != 1 {
+		return fmt.Errorf("query is not permitted to run - multi-statement query string")
+	}
+	for _, rawStmt := range parseResult.Stmts {
```

Comment:
I got a bit confused by this range (it almost looks like it's doing a tree walking while it's not doing).
Maybe better to do `rawStmt := parseResult.Stmts[0]`?

Reply:
Good idea, I made that change, and clarified the treewalk code so its easier to understand.